### PR TITLE
UI: Custom messages vault ui docs

### PR DESCRIPTION
**Pull request by kiannaquach**
🔍 [Deploy preview](https://vault-sxwma05dc-hashicorp.vercel.app/vault/docs/ui/custom-messages)

screencapture-localhost-3000-vault-docs-ui-custom-messages-2024-02-21-13_16_50

### CODE REVIEW

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [inline website/content/docs/ui/custom-messages/index.mdx:16]: I'm not sure if we use the terminology "Vault operator" or admin to refer to folks who manage Vault.

**Review by kiannaquach — COMMENTED** [review]
I added screenshots to the docs, and wanted to get thoughts on whether these screenshots would be helpful. If they are, then I'll create a new set of screenshots that don't include the localhost banner at the bottom of the screenshots. I'm also open to suggestions on what I should include in the screenshots like arrows, text, etc.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [inline website/content/docs/ui/custom-messages/index.mdx:47]: Are there any fields that *can't* be edited on a message either before or after it has been published/released?

If the answer is "no", then I'd suggest deleting this sentence.

If the answer is "yes", then I'd rewrite this to address what information can/can't be edited and when it becomes immutable.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [inline website/content/docs/ui/custom-messages/index.mdx:49]:
```suggestion
## Delete a custom message
```

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [inline website/content/docs/ui/custom-messages/index.mdx:53]: Suggest rewriting this section to follow the pattern of the Edit section.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [inline website/content/docs/ui/custom-messages/index.mdx:62]:
```suggestion
```
Suggest moving this info up to the intro section so folks understand how users will interact with the messages before taking the time to make them

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx]: Thanks for the suggestion! I will delete this sentence since there aren't any fields that can't be edited.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [inline website/content/docs/ui/custom-messages/index.mdx:25]: @schavis I'm not sure if multiple notes is too much information for the user, should we make the notes into bullet points instead?

We want to inform them that:
1. Messages are inheritable.
2. There can only be a maximum of 100 messages created. If it exceeds more than 100 message, they will need to delete one message before creating another.
3. It is recommended to avoid having more than one active modal at once since it can be intrusive for users.

Could you help with the language of these notes?

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [inline website/content/docs/ui/custom-messages/index.mdx:11]: @schavis we discussed adding a sentence or two that messages created aren't permanently.

Use custom banners and modals in the Vault UI to share system-wide alerts for all Vault UI users. Users can then dismiss messages by clicking "X" on a banner or "Confirm" for a modal. Clicking dismiss will not permanently dismiss the banner or modal. Users will see the message if they refresh the page or log out of their current session.
Vault UI sidebar. The "Custom Messages" button in Settings section is outlined with a red box

File: website/content/docs/ui/custom-messages/index.mdx
Comment (jasonodonnell) [inline website/content/docs/ui/custom-messages/index.mdx:16]: Can we define what permissions are needed? When testing this it seems to be https://developer.hashicorp.com/vault/api-docs/system/config-ui.

Comment (schavis) [reply website/content/docs/ui/custom-messages/index.mdx:16]: +1 to adding the more granular permissions if we can be specific

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [reply website/content/docs/ui/custom-messages/index.mdx:25]: Generally, multiple notes are bad because it trains folks to visually skip over them. Since this is general info about messages and not specific to creating them, I suggest moving the information about inheritability and size limit out of the aside boxes and making it part of the introduction to the page.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx:16]: Good catch! The [api docs for custom messages](https://github.com/hashicorp/vault/pull/24845/files) hasn't been published yet but will link to this page.

@schavis How do we typically define what permissions are needed?

File: website/content/docs/ui/custom-messages/index.mdx
Comment (schavis) [reply website/content/docs/ui/custom-messages/index.mdx:25]: I added a suggestion for new text in the intro :)

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx]: This is great! @marcboudreau can you confirm that create and edit require create permissions and edit does not require not update? I think we had a conversation in the past that create and edit are the same thing, but I wanted to be sure.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (marcboudreau) [inline website/content/docs/ui/custom-messages/index.mdx:26]: Given that the previous tip is about namespaces, it might be a good idea to change `100 messages` to `100 messages per namespace` here since, it would be easy to make the assumption that the 100 limit is global.

**Review by marcboudreau — APPROVED** [review]
Looks great! I left a comment to make a optional suggestion.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (Monkeychip) [inline website/content/docs/ui/custom-messages/index.mdx:44]: should these numbers be increasing?

File: website/content/docs/ui/custom-messages/index.mdx
Comment (Monkeychip) [inline website/content/docs/ui/custom-messages/index.mdx:72]: Same comment about numbers increasing.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (Monkeychip) [inline website/content/docs/ui/custom-messages/index.mdx:94]: Same comment about numbers increasing.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (Monkeychip) [inline website/content/docs/ui/custom-messages/index.mdx:99]: And here.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx:44]: The standard convention for markdown is to keep it at `1.` and it will auto format those numbered bullets to be increasing :)

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx]: Great idea!

File: website/data/docs-nav-data.json
Comment (schavis) [inline website/data/docs-nav-data.json:1233]: Typically, index pages are used for overviews related to the parent directory. Unless we expect to have many docs around custom messages, we should simplify the doc structure here and just have the `ui` folder with a `custom-messages.mdx` file underneath. Unless we have enough content/time to write a brief overview on the Vault UI, we can drop the index page for now.

File: website/content/docs/ui/custom-messages/index.mdx
Comment (yhyakuna) [reply website/content/docs/ui/custom-messages/index.mdx:44]: @Monkeychip This complies with the [markdown style guide](https://github.com/hashicorp/engineering-docs/blob/main/writing/markdown.md) as-is, so this is fine.

**Review by yhyakuna — APPROVED** [review]
LGTM 👍

File: website/content/docs/ui/custom-messages/index.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages/index.mdx]: I tested this locally and the policies that are needed are:

```
path "sys/config/ui/custom-messages" {
    capabilities = ["list", "create"]
}

path "sys/config/ui/custom-messages/*" {
    capabilities = ["read"]
}
```

File: website/content/docs/ui/custom-messages.mdx
Comment (marcboudreau) [inline website/content/docs/ui/custom-messages.mdx:36]: There are 2 different endpoints at play and the permissions needed are different.

In order to:
* **create or edit messages**, you must have the `list` and `create` permissions for the `sys/config/ui/custom-messages` endpoint and the `read` and `update` permissions for the `sys/config/ui/custom-messages/:id` endpoint
* **delete messages**, you must have the `list` permission for the `sys/config/ui/custom-messages` endpoint and the `delete` permission for the `sys/config/ui/custom-messages/:id` endpoint.

**Review by marcboudreau — CHANGES_REQUESTED** [review]
The stated permissions aren't quite correct.

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages.mdx]: Just updated!

File: website/content/docs/ui/custom-messages.mdx
Comment (marcboudreau) [inline website/content/docs/ui/custom-messages.mdx:35]: Maybe this line should be generalized to something like: `- **You must have the appropriate permissions**:`

Because the endpoint quoted here, doesn't actually exists. It's `custom-messages` and `custom-messages/:id`

**Review by marcboudreau — CHANGES_REQUESTED** [review]

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages.mdx]: Thanks for catching that!

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [inline website/content/docs/ui/custom-messages.mdx:38]: @schavis This was changed based on some feedback. Open to more feedback :)

File: website/content/docs/ui/custom-messages.mdx
Comment (schavis) [inline website/content/docs/ui/custom-messages.mdx:37]:
```suggestion
- **You must have the appropriate permissions**:
  - You must have `list` permission for the `sys/config/ui/custom-messages` endpoint.
  - To **create or edit messages**, you must have `read` and `update` permissions
    for the `sys/config/ui/custom-messages/:id` endpoint
  - To **delete messages**, `delete` permission for the
    `sys/config/ui/custom-messages/:id` endpoint.
```

File: website/content/docs/ui/custom-messages.mdx
Comment (schavis) [reply website/content/docs/ui/custom-messages.mdx:38]: Added a suggestion to make the information a little easier to scan.

**Review by schavis — APPROVED** [review]
Approving so you won't have to circle back for signoff from me, but please be sure to merge the requested readability changes before merging.

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages.mdx:45]: Yes that's correct the sidebar is "Custom Messages" and the page title is "Custom messages"

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages.mdx:25]: I'm a little confused here - in order to configure a custom message in a namespace, users will follow the same steps listed in `Add a custom message`

File: website/content/docs/ui/custom-messages.mdx
Comment (kiannaquach) [reply website/content/docs/ui/custom-messages.mdx:59]: From my understanding it's the browser's local timezone, however, in the [mdn docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local) for the date picker it says "local timezone".

File: website/content/docs/ui/custom-messages.mdx
Comment (yhyakuna) [reply website/content/docs/ui/custom-messages.mdx:25]: @finnstech  [**Namespaces consideration** section was added](https://vault-sxwma05dc-hashicorp.vercel.app/vault/docs/ui/custom-messages#namespaces-consideration) under the **Add a custom message** to explain the inheritance with an example.

**Comment by kiannaquach** [comment]
Closing in favor of this branch: https://github.com/hashicorp/vault/pull/25638

**Pull request closed without merging**
